Automattic/woocommerce-payments
# Refactor the `Tooltip` component to use TypeScript

#### Changes proposed in this Pull Request









This PR refactors the `Tooltip` and `TooltipBase` components into TypeScript, along with associated tests in this component directory. 

Refactoring these components to TypeScript will allow the extensibility of this component for the Simplify Deposits UI project as well as improve code quality.

No changes to accepted props or component behaviour are expected.

#### Testing Instructions

1. Ensure all unit and e2e tests pass.
2. Run `npx tsc --noEmit` to ensure no TS errors occur.
3. Visit one of the following screens to check if tooltips are rendered correctly.
  - Settings → Express Checkout
  - Transactions List → transaction amount, if currency conversion has occurred
  - Settings → Additional Payment Methods

     <img width="691" alt="image" src="https://user-images.githubusercontent.com/3147296/225788247-28a5496c-54f4-4fdb-8a7f-e1262fbd8123.png">

  - Settings → Additional Payment Methods (wizard that appears on first enable)
	
     <img width="592" alt="image" src="https://user-images.githubusercontent.com/3147296/225788157-c23d6fef-acae-42fe-8a93-5def8761ea0d.png">

-------------------


**Post merge**
